git_machete/client.py

```diff
@@ -778,6 +784,10 @@ def traverse(
                 else:
                     ans = self.ask_if(f"Rebase {bold(branch)} onto {bold(upstream)}?" + get_pretty_choices('y', 'N', 'q', 'yq'),
                                       f"Rebasing {bold(branch)} onto {bold(upstream)}...", opt_yes=opt_yes)
+
+                rebase = True
+                if branch in self.annotations:
+                    rebase = self.annotations[branch].qualifiers.rebase
```

Comment:
Whoops this should be checked **before** the question whether to rebase is asked! otherwise it's very confusing (the question is asked, and the rebase doesn't happen anyway) 🤔